src/asm/fixpoint.c

```diff
@@ -19,17 +19,23 @@
 #include "asm/symbol.h"
 #include "asm/warning.h"
 
-#define fix2double(i)	((double)((i) / 65536.0))
-#define double2fix(d)	((int32_t)round((d) * 65536.0))
-
-// pi radians == 32768 fixed-point "degrees"
-#define fdeg2rad(f)	((f) * (M_PI / 32768.0))
-#define rad2fdeg(r)	((r) * (32768.0 / M_PI))
-
 #ifndef M_PI
 #define M_PI 3.14159265358979323846
 #endif
 
+#ifndef M_TAU
+#define M_TAU (M_PI * 2)
+#endif
+
+#define fix2double(i)	((double)((i) / fix_PrecisionFactor()))
+#define double2fix(d)	((int32_t)round((d) * fix_PrecisionFactor()))
+
+// tau radians == 2**fixPrecision fixed-point "degrees"
+#define fdeg2rad(f)	((f) * M_TAU / fix_PrecisionFactor())
+#define rad2fdeg(r)	((r) * fix_PrecisionFactor() / M_TAU)
+
+int32_t fixPrecision;
```

Comment:
Why is it signed? Also, `uint8_t` should suffice as a range, and will be known safe for casting to `int` in `printf`.

Reply:
I agree, I was copying `fillByte`, but both could be refactored to `uint8_t`.